templates/shared/context.go

```diff
@@ -131,6 +131,10 @@ func Render(tpl *template.Template) func(ctx RuleContext) (string, error) {
 }
 
 func resolveRules(typ interface{ IsEmbed() bool }, rules *validate.FieldRules) (ruleType string, rule proto.Message, wrapped bool) {
+	if rules.GetMessage().GetSkip() {
```

Comment:
hm in my opinion, given the change we've made, it makes sense to keep this function as having just the oneof switch. 

I like putting the MessageRules on the RuleContext (in the body of rulesContext fun). 

Let's talk about it if you disagree.

Reply:
Then we can also eliminate all the HasRequiredAnnotation's

Reply:
Also I think it makes it simple to implement the other languages.

Reply:
The HasRequiredAnnotation sort of breaks the expectation that the rule resolution would be done by the time the template is executed.